### ISSUE TITLE
rule: `ban-type-builtin`

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#204]
## Description
`@Type` will be removed from Zig completely at some point, being replaced by `@Int`, `@Struct`, etc.
 https://github.com/ziglang/zig/issues/10710
Please describe what syntax or coding patterns this rule prohibits. Make sure you describe _why_ its bad.

## Examples

Examples of **incorrect** code for this rule
```zig
const Foo = @Type(.Int);
```

Examples of **correct** code for this rule
```zig
const Foo = @Int();
```
